src/json_writer.rs

```diff
@@ -4,14 +4,20 @@ use ::zip;
 use std::collections::HashMap;
 use std::fs;
 
-pub fn zips_to_json(file_paths: &[&str], exclude: &str) -> String {
-    let multi_archive = MultiArchiveJsonWriter::from(file_paths, exclude);
-    serde_json::to_string(&multi_archive).unwrap()
+use zip_info::StatArgs;
+
+pub fn zips_to_json(file_paths: &[&str], exclude: &str, stat_args: &StatArgs) -> String {
+    zips_to_json_with_printer(file_paths, exclude, stat_args, serde_json::to_string)
+}
+
+pub fn zips_to_json_pretty(file_paths: &[&str], exclude: &str, stat_args: &StatArgs) -> String {
+    zips_to_json_with_printer(file_paths, exclude, stat_args, serde_json::to_string_pretty)
 }
 
-pub fn zips_to_json_pretty(file_paths: &[&str], exclude: &str) -> String {
-    let multi_archive = MultiArchiveJsonWriter::from(file_paths,exclude);
-    serde_json::to_string_pretty(&multi_archive).unwrap()
+fn zips_to_json_with_printer<F>(file_paths: &[&str], exclude: &str, stat_args: &StatArgs, printer: F) -> String 
+    where F: Fn(&MultiArchiveJsonWriter) -> serde_json::Result<String> {
```

Comment:
I accidentally hit "start review" instead of just "single comment"--note that there are still comments, too